starport/cmd/network_chain_show.go

```diff
@@ -0,0 +1,238 @@
+)
+
+// NewNetworkChainShow creates a new chain show command to show
+// a chain on SPN.
+func NewNetworkChainShow() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "show [info|genesis|accounts|validators|peers] [launch-id]",
+		Short: "Show details of a chain",
+		RunE:  networkChainShowHandler,
+		Args:  cobra.ExactArgs(2),
+	}
+
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+
+	return c
+}
+
+func networkChainShowHandler(cmd *cobra.Command, args []string) error {
+	showType := ShowType(args[0])
+	if _, ok := showTypes[showType]; !ok {
+		cmd.Usage()
+		return fmt.Errorf("invalid arg %s", showType)
+	}
+
+	nb, err := newNetworkBuilder(cmd)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	// parse launch ID.
+	launchID, err := network.ParseLaunchID(args[1])
+	if err != nil {
+		return err
+	}
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	chainLaunch, err := n.ChainLaunch(cmd.Context(), launchID)
+	if err != nil {
+		return err
+	}
+
+	content := ""
+	switch showType {
+	case chainShowGenesis:
+		content, err = formatChainGenesis(nb, chainLaunch)
+	case chainShowInfo:
+		content, err = formatChainInfo(cmd.Context(), chainLaunch)
+	case chainShowAccounts:
+		content, err = formatChainAccounts(cmd.Context(), n, launchID)
+	case chainShowValidators:
+		content, err = formatChainValidators(cmd.Context(), n, launchID)
+	case chainShowPeers:
+		content, err = formatChainPeers(cmd.Context(), n, launchID)
+	}
+	if err != nil {
+		return err
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(content)
+	return nil
+}
+
+func formatChainGenesis(nb NetworkBuilder, chainLaunch networktypes.ChainLaunch) (string, error) {
+	c, err := nb.Chain(networkchain.SourceLaunch(chainLaunch))
+	if err != nil {
+		return "", err
+	}
+
+	genesisPath, err := c.GenesisPath()
+	if err != nil {
+		return "", err
+	}
+	// TODO run the prepare command in dry run mode and print the genesis if not exist
+	if _, err = os.Stat(genesisPath); os.IsNotExist(err) {
+		return "", errors.New("chain genesis not found, run the 'network chain prepare' command first")
+	}
+	genesisFile, err := os.ReadFile(genesisPath)
+	if err != nil {
+		return "", err
+	}
+	return string(genesisFile), nil
+}
+
+func formatChainInfo(ctx context.Context, chainLaunch networktypes.ChainLaunch) (info string, err error) {
+	var genesis []byte
+	if chainLaunch.GenesisURL != "" {
+		genesis, _, err = cosmosutil.GenesisAndHashFromURL(ctx, chainLaunch.GenesisURL)
+		if err != nil {
+			return "", err
+		}
+	}
+	chainInfo := struct {
+		Chain   networktypes.ChainLaunch `json:"Chain"`
+		Genesis []byte                   `json:"Genesis"`
+	}{
+		Chain:   chainLaunch,
+		Genesis: genesis,
+	}
+	return yaml.Marshal(ctx, chainInfo, "$.Genesis")
+}
+
+func formatChainAccounts(ctx context.Context, n network.Network, launchID uint64) (string, error) {
+	accountSummary := bytes.NewBufferString("")
```

Comment:
```suggestion
```

same for others

Reply:
https://github.com/tendermint/starport/pull/1892#discussion_r768961055